### PR TITLE
Add multiplayer logic

### PR DESCRIPTION
**Pull request by khooroko**
**Description**
ADD MULTIPLAYER!!!!!!!!!
(also added a Utils file)

@lyhvictoria

**Impact**
- [x] **MAJOR**

**Risks**
Huge refactor of objects, and possibly stinky code left in some places forgotten to clean due to desperation. Will clean eventually but what's important for now is it works. Completely breaks `MainScene` but should be fine after renaming `MultiplayerTestScene` to `MainScene`.

**Test Plan**
Test with 2 players then with one player (both with and without going through lobby). All core functionality should still be intact. Also should test for things added in #26 because I might have screwed up bits and pieces of the rebase

### CODE REVIEW

File: Assets/Scripts/NetworkController.cs [hunk 1]
```diff
@@ -31,7 +31,10 @@ public class NetworkController : MonoBehaviourPunCallbacks {
 	public Button LeaveRoomButton;
 	public GameObject PlayerListEntryPrefab;
 
-	private Dictionary<string, RoomInfo> cachedRoomList;
+    [Header("RoomManager")]
+    public GameObject RoomManagerPrefab;
+
+    private Dictionary<string, RoomInfo> cachedRoomList;
 	private Dictionary<string, GameObject> roomListEntries;
```
Comment (lyhvictoria) [inline Assets/Scripts/NetworkController.cs]: spacing here

File: Assets/Scripts/PlayerManager.cs [hunk 2]
```diff
@@ -1,17 +1,4 @@
...
-    public GameObject HumanPrefab;
-    public GameObject BotPrefab;
-    public GameObject OffScreenUIPrefab;
-    // Controls whether we need to spawn player 1
-    public bool isPlayerInScene = true;
-    public Transform player1SpawnPoint;
-    public Transform player2SpawnPoint;
-    public ScoreBoard[] scoreBoards;
-    private GameObject player1, player2;
+﻿public static class Util {
```
Comment (lyhvictoria) [inline Assets/Scripts/PlayerManager.cs]: Might as well either rename this script to utils or something. To be addressed next time.

**Review by lyhvictoria — APPROVED** [review]
There's a Utils class in PlayerManager and also a Utils.cs. That might get confusing in future. Otherwise looks good.

File: Assets/Scripts/PlayerManager.cs (hunk 2 above)
Comment (khooroko) [reply Assets/Scripts/PlayerManager.cs]: o i forgot to rename the file oops

**Pull request merged**